source/AlarmWindow.cpp

```diff
@@ -50,72 +51,51 @@ AlarmWindow :: AlarmWindow (BRect frame, BHandler *handler)
     fMessenger = new BMessenger(handler);
 
     char	dayDefaultField[3];
-    char 	monthDefaultField[3];
-    char 	yearDefaultField[3];
-    char 	minuteDefaultField[3];
-    char 	hourDefaultField[3];
+    char 	monthDefaultField[4];
+    char 	yearDefaultField[5];
+    char 	hourNow[6];
+	char	minNow[7];
 
 	// Initialize text fields with current system time values
-	sprintf(minuteDefaultField, "%02d", GetTime(0));
-	sprintf(hourDefaultField, "%02d", GetTime(1));
+	sprintf(minNow,"%02d", GetTime(0)+3);
 	sprintf(dayDefaultField, "%02d", GetTime(2));
+	sprintf(hourNow, "%02d", GetTime(1));
 	sprintf(monthDefaultField, "%02d", GetTime(3));
 	sprintf(yearDefaultField, "%d", GetTime(4));
 
 	// We allocate the view AlarmView and associate it to the AlarmWindow
-	frame.OffsetTo(B_ORIGIN);
-	fAlarmView = new AlarmView(frame,"AlarmView");
-	fAlarmView->SetViewColor(ui_color(B_PANEL_BACKGROUND_COLOR));
-	//AddChild(fAlarmView);
-
 	// Text fields for the data
-	hour = new BTextControl(BRect(10,10,100,35),   B_TRANSLATE("hour"),     B_TRANSLATE("hour:"), hourDefaultField , NULL);
-	minute = new BTextControl(BRect(110,10,210,35), B_TRANSLATE("minute"),  B_TRANSLATE("min:"), minuteDefaultField, NULL);
-	day = new BTextControl(BRect(10,50,100,75),   B_TRANSLATE("day"),  B_TRANSLATE("day:"), dayDefaultField, NULL);
-	month = new BTextControl(BRect(110,50,200,75), B_TRANSLATE("month"),    B_TRANSLATE("month:"), monthDefaultField, NULL);
-	year = new BTextControl(BRect(210,50,300,75),  B_TRANSLATE("year"),     B_TRANSLATE("year:"), yearDefaultField, NULL);
-
-	// Text field label: visible
-	hour -> SetDivider(hour->StringWidth("hour:") + 5);
-	minute -> SetDivider(minute->StringWidth("min:") + 5);
-	day -> SetDivider(day->StringWidth("day:") + 5);
-	month -> SetDivider(month->StringWidth("month:") + 5);
-	year -> SetDivider(year->StringWidth("year:") + 5);
+	hour = 		new BTextControl("hour",    B_TRANSLATE("hour:"), hourNow , NULL);
```

Comment:
That's a bitof a weird indentation, isn't it? Mixing spaces and tabs... Why not use normal coding style with just one space after = and ,?
Also, I think those labels look nicer when capitalized, and having the unabbrev. "Minutes:".

Depending how much effort you wanna invest, for the date you could very well use a BCalendarView, see
haiku/headers/private/shared/CalendarView.h
The time might use a BSpinner for easy in/decreasing.

But as I said, I'm OK if you don't feel like sinking time into that... :)